iron-image.html

```diff
@@ -298,65 +330,43 @@
         this.style.height = isNaN(this.height) ? this.height : this.height + 'px';
       },
 
+      _preventLoadChanged: function() {
+        if (this.preventLoad || this.loaded) return;
+
+        this._reset();
+        this._load(this.src);
+      },
+
       _srcChanged: function(newSrc, oldSrc) {
-        if (newSrc !== oldSrc) {
-          this.loaded = false;
+        if (newSrc === oldSrc) return;
```

Comment:
Potentially you can use the same `this._matchesSrc` trick here and not reset if the new src is effectively the same once resolved..